File: 01-basic/T01-fibonacci/fib.py
def fib(N): # defines function "fib" with inpara "N"
    assert(N > 2)
    # p_1, p_2 = 1, 1
    p_1 =1
    p_2 =1
    print(p_1)
    print(p_2)
    for _ in range(2, N):    # starts a while loop till "cnt" hits "N"
        p_1, p_2 = p_1 + p_2, p_1
        print(p_1)

fib(10)

File: 01-basic/T01-fibonacci/Fib.c
#include <stdio.h>
#include <assert.h>

void fib(int N)
{
    int p_1 = 1, p_2 = 1, p, cnt;
    assert(N>2);
    printf("%d\n%d\n", p_1, p_2);
    cnt = 2;
    while (cnt < N) {
        p = p_1 + p_2;
        p_2 = p_1;
        p_1 = p;
        printf("%d\n", p);
        ++cnt;
    }
}

int main(int argc, char* argv[])
{
    fib(10);
    getc(stdin);
    return 0;
}
